galsim/gsobject.py

```diff
@@ -1713,6 +1721,9 @@ def drawPhot(self, image, n_photons=0, rng=None, max_extra_noise=None, poisson_f
 
             phot_array.scaleFlux(g * thisN / Ntot)
 
+            for op in surface_ops:
```

Comment:
I was hoping that the constructors for these operators would be able to get any necessary information such as where in the focal plane this postage stamp sits or where the telescope is pointing, etc. outside of drawImage.  Some could be set in the constructor.  And some could be done as setters that would be called before drawImage.  

The config layer has similar hooks for things that need to be done at the start of an image or at the start of a stamp for other kinds of things.  Of course these surface ops aren't in there yet, but I think it will be straightforward to do the same thing for these.  

And hopefully that would cover all use cases, but of course we may need to revisit if something doesn't fit this model.